miner/src/pool/listener.rs

```diff
@@ -116,6 +116,70 @@ impl txpool::Listener<Transaction> for Logger {
 	}
 }
 
+/// Transactions pool notifier
+#[derive(Default)]
+pub struct TransactionsPoolNotifier {
+	listeners: Vec<Arc<Mutex<mpsc::Sender<(H256, String)>>>>,
+}
+
+impl TransactionsPoolNotifier {
+	/// Add new listener to receive notifications.
+	pub fn add(&mut self, f: mpsc::Sender<(H256, String)>) {
+		self.listeners.push(Arc::new(Mutex::new(f)));
+	}
+
+	/// Notify listeners about all currently transactions.
+	fn notify(& mut self, hash: H256, status: String) {
+		for l in &self.listeners {
+			let l = l.lock();
+			if let Ok(l) = l {
+				let pair = (hash.clone(), status.clone());
+				let _ = l.send(pair);
+			}
+		}
+	}
+}
+
+impl fmt::Debug for TransactionsPoolNotifier {
+	fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+		fmt.debug_struct("TransactionsPoolNotifier")
+			.field("listeners", &self.listeners.len())
+			.finish()
+	}
+}
+
+impl txpool::Listener<Transaction> for TransactionsPoolNotifier {
+	fn added(&mut self, tx: &Arc<Transaction>, _old: Option<&Arc<Transaction>>) {
+		let hash = tx.hash.clone();
+		self.notify(hash, "added".to_string());
```

Comment:
I think it would be better to have an enum like this:
```
#[derive(Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
enum Status {
 Added,
 Rejected,
 ...
 Culled,
}
```
And the same status enum should be sent in the notification.